src/tdastro/sources/bayeSN_model.py

```diff
@@ -0,0 +1,493 @@
+import h5py
+import numpy as np
+from astropy import units as u
+
+from tdastro.astro_utils.unit_utils import flam_to_fnu
+from tdastro.base_models import FunctionNode
+from tdastro.effects.extinction import ExtinctionEffect
+from tdastro.sources.physical_model import PhysicalModel
+
+
+class bayeSN_Model(PhysicalModel):
```

Comment:
Can you make the class for citation.  This has two additional steps:
1) Add "from citation_compass import CiteClass"
2) Change "class bayeSN_Model(PhysicalModel):" -> "class bayeSN_Model(PhysicalModel, CiteClass):"

You already have the references section in your docstring, so the actual citations will be picked up automatically.